src/app/common/account-restoration/legacy-gaia-config-lookup.ts

```diff
@@ -0,0 +1,20 @@
+import { fetchWalletConfig, generateWallet } from '@stacks/wallet-sdk';
+import { connectToGaiaHubWithConfig, getHubInfo } from '@stacks/wallet-sdk';
+
+import { gaiaUrl } from '@shared/constants';
+
+export async function checkForLegacyGaiaConfigWithKnownGeneratedAccountIndex(secretKey: string) {
```

Comment:
Will we always have to keep this legacy check?

Reply:
Replaced this with a look up directly to gaia config. The old method was firing a bns req. per account, potentially hundreds every time.

To avoid any visible functional changes, reckon it makes sense to leave as is for now.